bindings/python/deckgl-widget/js/lib/extension-core.js

```diff
@@ -0,0 +1,174 @@
+}
+
+function createCanvas(idName) {
+  var canvas = document.createElement('canvas');
+  canvas.style.width = '100%';
+  canvas.style.height = '100%';
+  canvas.style.position = 'absolute';
+  canvas.style['z-index'] = 2;
+  canvas.id = idName;
+  return canvas;
+}
+
+function createMapDiv() {
+  var div = document.createElement('div');
+  div.id = 'map';
+  div.style['pointer-events'] = 'none';
+  div.style['height'] = '100%';
+  div.style['width'] = '100%';
+  div.style['position'] = 'absolute';
+  div.style['z-index'] = 1;
+  return div;
+}
+
+require.config({
+  paths: {
+    deckglbase: 'https://unpkg.com/deck.gl@6.4.5/deckgl.min',
+    mapboxgl: 'https://api.tiles.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl',
+    d3dsv: 'https://cdnjs.cloudflare.com/ajax/libs/d3-dsv/1.0.8/d3-dsv.min',
+    deckJson: 'https://duberste.in/deckJson'
+  },
+  shim: {
+    deckglbase: {
+      exports: 'deckglbase',
+      deps: ['mapboxgl']
+    },
+    mapboxgl: {
+      exports: 'mapboxgl',
+    },
+    deckJson: {
+      deps: ['deckglbase', 'd3dsv'],
+      init: function(deckbase, d3dsv) {
+        var exports = {};
+        var deckJsonApiExports = deckJsonApi(exports, deckbase, d3dsv);
+        return {
+          'JSONConverter': deckJsonApiExports._JSONConverter}
+      },
+      exports: 'deckJson'
+    }
+  }
+});
+
+
+define(
+  'deckwidget', 
+  ['@jupyter-widgets/base', 'deckglbase', 'mapboxgl', 'deckJson'],
+  function(widgets, deckgl, mapboxgl, deckJson) {
+
+    function setMapProps(map, props) {
+      if ('viewState' in props && props.viewState.longitude && props.viewState.latitude) {
+        const {viewState} = props;
+        map.jumpTo({
+          center: [viewState.longitude, viewState.latitude],
+          zoom: viewState.zoom || 10,
+          bearing: viewState.bearing || 0,
+          pitch: viewState.pitch || 0
+        });
+      }
+    
+      if (props.map && 'style' in props.map) {
+        if (props.map.style !== map.deckStyle) {
+          map.setStyle(props.map.style);
+          map.deckStyle = props.map.style;
+        }
+      }
+    }
+
+    function hideMapboxCSSWarning() {
+      var missingCssWarning = document.getElementsByClassName('mapboxgl-missing-css')[0];
+      if (missingCssWarning) {
+        missingCssWarning.style.display = 'none';
+      }
+    }
+
+    var DeckView = widgets.DOMWidgetView.extend({
+
+       _onViewStateChange: function({viewState}) {
+          this.deck.setProps({viewState});
+          setMapProps(this.mapLayer, {viewState})
+      },
+
+      render: function() {
+          this.model.on('change:json_input', this.value_changed, this);
+          var mapNode = createMapDiv()
+          var canvasNode = createCanvas('deck-map-container')
+          this.el.appendChild(createDiv('deck-container'))
+                 .appendChild(createDiv('deck-map-wrapper'))
+                 .appendChild(canvasNode);
+          canvasNode.parentNode.insertBefore(mapNode, canvasNode.nextSibling);
```

Comment:
You can add logic here to check whether `model.get('json_input')` is truthy, and call `this.createDeck` with the existing value. Doing so will ensure that the map re-rendered when the notebook is reloaded